### ISSUE TITLE
Starter Upgrade failing with "no route to host" error message

### ISSUE THREAD

**Issue by garyhampson** [arangodb/arangodb#15498]
## My Environment

* __ArangoDB Version__:        3.7.13 
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster 
* __Deployment Strategy__:   ArangoDB Starter
* __Configuration__:               3 node cluster
* __Infrastructure__:               Azure
* __Operating System__:        CentOS Linux 7
* __Total RAM in your machine__:        16 GiB
* __Disks in use__:        SSD 2048 GiB
* __Used Package__:              RedHat .rpm 

## Component, Query & Data

__Affected feature__:
Upgrade 

__Size of your Dataset on disk__:
135 GiB

__Replication Factor & Number of Shards (Cluster only)__:
3 & 1 for all collections

## Steps to reproduce

1. Run the following to install the upgrade on each cluster node:
```
sudo su
rundate="$(date +%Y%m%d)"
sed -i.$rundate 's/exclude=arango/#exclude=arango/' /etc/yum.conf
mv /etc/yum.repos.d/arangodb.repo /etc/yum.repos.d/arangodb.repo.$(date +'%Y%m%d')
touch /etc/yum.repos.d/arangodb.repo
echo "[arangodb]
name=ArangoDB Project
type=rpm-md
baseurl=https://download.arangodb.com/arangodb38/RPM
gpgcheck=1
gpgkey=https://download.arangodb.com/arangodb38/RPM/repodata/repomd.xml.key
enabled=1" >> /etc/yum.repos.d/arangodb.repo
yum clean all
yum install arangodb3-3.8.4-1.0 -y
sed -i 's/Restart=on-failure/#Restart=on-failure/' /usr/lib/systemd/system/arangodb3cluster.service
sed -i 's/#KillMode=process/KillMode=process/' /usr/lib/systemd/system/arangodb3cluster.service
systemctl daemon-reload
systemctl status arangodb3cluster.service -l
kill -9 $(ps -C arangodb -fww --no-headers | awk '{print $2}')
systemctl status arangodb3cluster.service
systemctl start arangodb3cluster.service
systemctl status arangodb3cluster.service -l
```
2. Run the following on a single node to start the upgrade process:
```
arangodb upgrade --starter.endpoint=http://$(ifconfig|grep inet |awk '{print $2}' | head -1):8528
```
Then run the following to finalize the upgrade:
```
sed -i 's/#Restart=on-failure/Restart=on-failure/' /usr/lib/systemd/system/arangodb3cluster.service
sed -i 's/KillMode=process/#KillMode=process/' /usr/lib/systemd/system/arangodb3cluster.service
systemctl daemon-reload
systemctl restart arangodb3cluster.service
sed -i 's/#exclude=arango/exclude=arango/' /etc/yum.conf
```
3. 

__Problem__:
2022-01-11T05:25:57Z |FATAL| Failed to start database automatic upgrade error="Get \"http://10.241.4.8:8528/version\": dial tcp 10.241.4.8:8528: connect: no route to host" component=arangodb

Not sure where 10.241.4.8 is coming from as that isn't specified in my starter configuration.

__Expected result__:
2022-01-11T03:56:43Z |INFO| Database automatic upgrade from version 3.7.13 to version 3.8.4 has been started component=arangodb
2022-01-11T03:56:43Z |INFO| Servers upgraded: none, remaining servers: 3 agents, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T03:56:50Z |INFO| Servers upgraded: 1 agent, remaining servers: 2 agents, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T03:57:06Z |INFO| Servers upgraded: 2 agents, remaining servers: 1 agent, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T03:57:07Z |INFO| Servers upgraded: 3 agents, remaining servers: 3 coordinators, 3 dbservers component=arangodb
2022-01-11T03:57:14Z |INFO| Servers upgraded: 3 agents, 1 dbserver, remaining servers: 3 coordinators, 2 dbservers component=arangodb
2022-01-11T03:57:22Z |INFO| Servers upgraded: 3 agents, 2 dbservers, remaining servers: 3 coordinators, 1 dbserver component=arangodb
2022-01-11T03:57:29Z |INFO| Servers upgraded: 3 agents, 3 dbservers, remaining servers: 3 coordinators component=arangodb
2022-01-11T03:57:36Z |INFO| Servers upgraded: 3 agents, 1 coordinator, 3 dbservers, remaining servers: 2 coordinators component=arangodb
2022-01-11T03:57:43Z |INFO| Servers upgraded: 3 agents, 2 coordinators, 3 dbservers, remaining servers: 1 coordinator component=arangodb
2022-01-11T03:57:52Z |INFO| Database upgrade has finished component=arangodb

Here are my configuration settings:

```
[root@dev-eastus-arangodb-elasticsearch-1 datadrive]# more /usr/lib/systemd/system/arangodb3cluster.service                                            
[Unit]
Description=Run the ArangoDB Starter
After=network.target

[Service]
#Restart=on-failure
EnvironmentFile=/etc/arangodb3/arangodb.cluster.env
ExecStart=/usr/bin/arangodb \
    --starter.address=${PRIVATEIP} \
    --starter.data-dir=${DATADIR} \
    --starter.join=${STARTERENDPOINTS} \
    --server.storage-engine=rocksdb
StandardOutput=syslog
StandardError=syslog
SyslogIdentifier=arango
TimeoutStopSec=60
KillMode=process

[Install]
WantedBy=multi-user.target
```

```
[root@dev-eastus-arangodb-elasticsearch-1 datadrive]# more /etc/arangodb3/arangodb.cluster.env                                                         
DATADIR=/var/lib/arangodb3/cluster
PRIVATEIP=10.241.4.4
STARTERENDPOINTS=10.241.4.4,10.241.4.5,10.241.4.6
```

**Comment by garyhampson** [comment]
So, I found that ip address in the setup.json file in our /datadrive/arangodb3/cluster (symlinked to /var/lib/arangodb3/cluster) directory and have removed it from all three nodes:
```
{"version":"0.2.2","id":"dc257820","peers":{"Peers":[{"ID":"dc257820","Address":"10.241.4.4","Port":8528,"PortOffset":0,"DataDir":"/var/lib/arangodb3/cluster","HasAgent":true,"IsSecu
re":false},{"ID":"4107bc33","Address":"10.241.4.5","Port":8528,"PortOffset":0,"DataDir":"/var/lib/arangodb3/cluster","HasAgent":true,"IsSecure":false},{"ID":"5bdc9bb8","Address":"10.
241.4.6","Port":8528,"PortOffset":0,"DataDir":"/var/lib/arangodb3/cluster","HasAgent":true,"IsSecure":false}],"AgencySize":3,"LastModified":"2021-11-26T10:14:00.786467266Z","PortOffs
etIncrement":10,"ServerStorageEngine":"rocksdb"},"mode":"cluster"}
```

Now when I try to start the starter, I get this:
```
2022-01-11T16:59:26Z |INFO| restarting agent component=arangodb pid=13079 type=agent
2022-01-11T16:59:26Z |INFO| Looking for a running instance of agent on port 8531 component=arangodb pid=13079 type=agent
2022-01-11T16:59:26Z |INFO| Starting agent on port 8531 component=arangodb pid=13079 type=agent
2022-01-11T16:59:26Z |INFO| server started component=arangodb pid=13212 type=agent
2022-01-11T16:59:27Z |INFO| Wait on 13212 returned component=arangodb exit-status=1 trap-cause=-1
2022-01-11T16:59:27Z |INFO| Terminated agent component=arangodb pid=13212 type=agent
2022-01-11T16:59:27Z |INFO| agent has terminated quickly, in 827.608313ms (recent failures: 55) component=arangodb pid=13212 type=agent
2022-01-11T16:59:27Z |INFO| ## Start of agent log
        2022-01-11T16:59:26Z [13212] INFO [25362] {memory} Available physical memory: 16825786368 bytes, available cores: 4
        2022-01-11T16:59:26Z [13212] WARNING [e8b68] {memory} /sys/kernel/mm/transparent_hugepage/enabled is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
        2022-01-11T16:59:26Z [13212] WARNING [e8b68] {memory} /sys/kernel/mm/transparent_hugepage/defrag is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
        2022-01-11T16:59:26Z [13212] WARNING [f3108] {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/enabled"'
        2022-01-11T16:59:26Z [13212] WARNING [f3108] {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/defrag"'
        2022-01-11T16:59:26Z [13212] INFO [144fe] {general} using storage engine 'rocksdb'
        2022-01-11T16:59:26Z [13212] INFO [3bb7d] {cluster} Starting up with role AGENT
        2022-01-11T16:59:26Z [13212] INFO [f6e0e] {aql} memory limit per AQL query automatically set to 10095471821 bytes. to modify this value, please adjust the startup option `--query.memory-limit`
        2022-01-11T16:59:26Z [13212] INFO [a1c60] {syscall} file-descriptors (nofiles) hard limit is 65535, soft limit is 65535
        2022-01-11T16:59:27Z [13212] INFO [fe333] {engines} RocksDB recovery starting, scanning WAL starting from sequence number 7820839
        2022-01-11T16:59:27Z [13212] INFO [a4ec8] {engines} RocksDB recovery finished, WAL entries scanned: 372675, max tick found in WAL: 5004737, last HLC value: 1721676306722062337
        2022-01-11T16:59:27Z [13212] ERROR [3bc7f] {startup} Database directory version (30713) is lower than current version (30804).
        2022-01-11T16:59:27Z [13212] ERROR [ebca0] {startup} ----------------------------------------------------------------------
        2022-01-11T16:59:27Z [13212] ERROR [24e3c] {startup} It seems like you have upgraded the ArangoDB binary.
        2022-01-11T16:59:27Z [13212] ERROR [8bcec] {startup} If this is what you wanted to do, please restart with the
        2022-01-11T16:59:27Z [13212] ERROR [b0360] {startup}   --database.auto-upgrade true
        2022-01-11T16:59:27Z [13212] ERROR [13414] {startup} option to upgrade the data in the database directory.
        2022-01-11T16:59:27Z [13212] ERROR [24bd1] {startup} ----------------------------------------------------------------------'
        2022-01-11T16:59:27Z [13212] ERROR [1c156] {general} Database '_system' needs upgrade. Please start the server with --database.auto-upgrade
        2022-01-11T16:59:27Z [13212] FATAL [2eb08] {general} Database '_system' upgrade failed (bad parameter). Please inspect the logs from the upgrade procedure and try starting the server again.
```
which leads to this:
```
2022-01-11T17:03:25Z |FATAL| Failed to start database automatic upgrade error="Post \"http://10.241.4.4:8528/database-auto-upgrade\": dial tcp 10.241.4.4:8528: connect: connection refused" component=arangodb
```

**Comment by garyhampson** [comment]
And now a `yum remove arangodb3-3.8.4-1.0 -y` and `yum install arangodb3-3.8.4-1.0 -y` with all the attendant other commands (`systemctl daemon-reload`, etc) seems to have worked allowing me to "upgrade" from 3.8.4 to 3.8.4:
```
[root@dev-eastus-arangodb-elasticsearch-1 cluster]# arangodb upgrade --starter.endpoint=http://$(ifconfig|grep inet |awk '{print $2}' | head -1):8528
2022-01-11T18:26:59Z |INFO| Database automatic upgrade from version 3.8.4 to version 3.8.4 has been started component=arangodb
2022-01-11T18:26:59Z |INFO| Servers upgraded: none, remaining servers: 3 agents, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T18:27:09Z |INFO| Servers upgraded: 1 agent, remaining servers: 2 agents, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T18:27:15Z |INFO| Servers upgraded: 2 agents, remaining servers: 1 agent, 3 coordinators, 3 dbservers component=arangodb
2022-01-11T18:27:17Z |INFO| Servers upgraded: 3 agents, remaining servers: 3 coordinators, 3 dbservers component=arangodb
2022-01-11T18:27:24Z |INFO| Servers upgraded: 3 agents, 1 dbserver, remaining servers: 3 coordinators, 2 dbservers component=arangodb
2022-01-11T18:27:31Z |INFO| Servers upgraded: 3 agents, 2 dbservers, remaining servers: 3 coordinators, 1 dbserver component=arangodb
2022-01-11T18:27:38Z |INFO| Servers upgraded: 3 agents, 3 dbservers, remaining servers: 3 coordinators component=arangodb
2022-01-11T18:27:44Z |INFO| Servers upgraded: 3 agents, 1 coordinator, 3 dbservers, remaining servers: 2 coordinators component=arangodb
2022-01-11T18:27:51Z |INFO| Servers upgraded: 3 agents, 2 coordinators, 3 dbservers, remaining servers: 1 coordinator component=arangodb
2022-01-11T18:27:57Z |INFO| Database upgrade has finished component=arangodb
```
